examples/some-verifier/src/db.rs

```diff
@@ -242,13 +253,45 @@ impl Database {
         let mut client = self.client.write().await;
         let transaction = client.transaction().await?;
 
-        let statement = format!(
+        // Clear pre-existing verifications with overlapping credentials;
+        let (usings, wheres, cred_ids) = Platform::SUPPORTED_PLATFORMS
+            .into_iter()
+            .filter_map(|p| entry.platform_entry(p).map(|e| (p, e)))
+            .enumerate()
+            .fold(
+                (String::new(), String::new(), Vec::new()),
+                |(mut usings, mut wheres, mut cred_ids), (i, (platform, entry))| {
+                    write!(usings, ", {}", platform.table_name()).expect("can write to String");
+                    write!(
+                        wheres,
+                        " OR ({0}.{VERIFICATION_ID_COLUMN}={VERIFICATIONS_TABLE}.{ID_COLUMN} AND {0}.{CRED_ID_COLUMN}=${1})",
+                        platform.table_name(),
+                        i + 1
+                    )
+                    .expect("can write to String");
+                    cred_ids.push(entry.cred_id.public_key.as_bytes() as &(dyn ToSql + Sync));
+                    (usings, wheres, cred_ids)
+                },
+            );
+
+        // Remove leading ','
+        let usings = usings.split_once(',').map(|(_, res)| res).unwrap_or("");
+        // Remove leading "OR"
+        let wheres = wheres.split_once(" OR").map(|(_, res)| res).unwrap_or("");
+
+        let delete_statement =
+            format!("DELETE FROM {VERIFICATIONS_TABLE} USING{usings} WHERE{wheres}");
+        let delete = transaction.execute(&delete_statement, &cred_ids);
+
+        let insert_statement = format!(
             "INSERT INTO {VERIFICATIONS_TABLE} ({}) VALUES ({}) RETURNING id",
             columns.join(", "),
             (1..=columns.len()).format_with(", ", |i, f| f(&format_args!("${i}")))
         );
+        let insert = transaction.query_one(&insert_statement, &values);
 
-        let verification_id: i64 = transaction.query_one(&statement, &values).await?.get(0);
+        // Run delete and insert, retrieve new verification id
+        let verification_id: i64 = try_join!(delete, insert)?.1.get(0);
```

Comment:
Does `try_join` run these sequentially? The documentation states that it polls them simultaneously, which i imagine is not what we want (if it means concurrently, that is)?

Reply:
Don't we always want to run delete first and only then we run insert?

So not just not concurrently, but in a specific order.

Reply:
Yes, what I mean is we don't want them to run in parallel, because we want to run them sequentially, in the same order every time (and I don't know if `try_join` runs futures in parallel).

Reply:
try_join will run them in parallel.

Reply:
I updated this.